Add docs on replica circuit breakers (#13676)
cockroachdb/docs: v22.1/architecture/replication-layer.md

This feature is designed to increase the availability of your CockroachDB clusters by making them more robust to transient errors.

For more information about how to view per-replica circuit breaker events happening on your cluster in the [DB Console](../ui-overview.html), see the [Replication Dashboard](../ui-replication-dashboard.html) documentation.

<a name="per-replica-circuit-breaker-configuration"></a>

##### Configuration

> Per-replica circuit breakers are enabled by default. Most users will not have to do any configuration to get the benefits of this feature.

Reviewer: ```suggestion
Per-replica circuit breakers are enabled by default. Most users will not have to configure anything to get the benefits of this feature.
```

<a name="per-replica-circuit-breaker-timeout"></a>

The circuit breaker timeout value is controlled by the `kv.replica_circuit_breaker.slow_replication_threshold` [cluster setting](../cluster-settings.html), which defaults to an [interval](../interval.html) of `1m0s` (1 minute).

<a name="per-replica-circuit-breaker-limitations"></a>

##### Limitations

Per-replica circuit breakers have the following limitations:

> - They cannot prevent hung requests when the node's [liveness range](#epoch-based-leases-table-data) is unavailable. For more information about troubleshooting a cluster that's having node liveness issues, see [Node liveness issues](../cluster-setup-troubleshooting.html#node-liveness-issues).

Reviewer: ```suggestion
- They cannot prevent requests from hanging when the node's [liveness range](#epoch-based-leases-table-data) is unavailable. For more information about troubleshooting a cluster that's having node liveness issues, see [Node liveness issues](../cluster-setup-troubleshooting.html#node-liveness-issues).
```
